src/main/java/com/redhat/devtools/lsp4ij/features/workspaceSymbol/AbstractLSPWorkspaceSymbolContributor.java

```diff
@@ -1,109 +1,133 @@
-            //TODO delete block when minimum required version is 2024.2
-            workspaceSymbolSupport.cancel();
-            return null;
-        } catch (CancellationException e) {
-            // cancel the LSP requests workspace/symbol
-            workspaceSymbolSupport.cancel();
-            return null;
-        } catch (ExecutionException e) {
-            LOGGER.error("Error while consuming LSP 'workspace/symbol' request", e);
-            return null;
-        }
-
-        if (isDoneNormally(workspaceSymbolFuture)) {
-            // workspace/symbol has been collected correctly
-            return workspaceSymbolFuture.getNow(null);
-        }
-        return null;
-    }
-
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.workspaceSymbol;
+
+import com.intellij.ide.util.gotoByName.ChooseByNamePopup;
+import com.intellij.navigation.ChooseByNameContributorEx;
+import com.intellij.navigation.NavigationItem;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.util.Processor;
+import com.intellij.util.indexing.FindSymbolParameters;
+import com.intellij.util.indexing.IdFilter;
+import com.redhat.devtools.lsp4ij.LSPWorkspaceSupport;
+import org.eclipse.lsp4j.WorkspaceSymbolParams;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.isDoneNormally;
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.waitUntilDone;
+
+/**
+ * Abstract base class for LSP goto symbol contributors
+ */
+abstract class AbstractLSPWorkspaceSymbolContributor implements ChooseByNameContributorEx {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(AbstractLSPWorkspaceSymbolContributor.class);
+
+    private final LSPWorkspaceRequestedSymbolTypes requestedSymbolTypes;
+
+    protected AbstractLSPWorkspaceSymbolContributor(@NotNull LSPWorkspaceRequestedSymbolTypes requestedSymbolTypes) {
```

Comment:
The concrete implementation of this must specify what types of symbols it is requesting, currently "All" vs. "Types".